framework/src/modules/chain/logic/state_store/account_store.js

```diff
@@ -125,12 +129,17 @@ class AccountStore {
 			})
 		);
 
-		return affectedAccounts.map(({ updatedItem, updatedKeys }) => {
-			const filter = { [this.primaryKey]: updatedItem[this.primaryKey] };
-			const updatedData = _.pick(updatedItem, updatedKeys);
+		const updateToAccounts = affectedAccounts.map(
+			({ updatedItem, updatedKeys }) => {
+				const filter = { [this.primaryKey]: updatedItem[this.primaryKey] };
+				const updatedData = _.pick(updatedItem, updatedKeys);
+				updatedData.u_balance = updatedData.balance;
+
+				return this.account.upsert(filter, updatedData, null, this.tx);
+			}
+		);
 
-			return this.account.upsert(filter, updatedData, null, tx);
-		});
+		return updateToAccounts;
```

Comment:
instead of requiring `Promise.all` in the function user, maybe Promise.all here?